core/backend/src/annotations/TextAnnotationElement.ts

```diff
@@ -154,35 +200,73 @@ export class TextAnnotation2d extends AnnotationElement2d /* implements ITextAnn
   }
 
   /**
-   * Updates the geometry of the TextAnnotation2d on insert.
+   * Updates the geometry of the TextAnnotation2d on insert and validates version.
    * @inheritdoc
    * @beta
    */
   protected static override onInsert(arg: OnElementPropsArg): void {
     super.onInsert(arg);
-    this.updateGeometry(arg.iModel, arg.props as TextAnnotation2dProps);
+    this.validateVersionAndUpdateGeometry(arg);
   }
 
   /**
-   * Updates the geometry of the TextAnnotation2d on update.
+   * Updates the geometry of the TextAnnotation2d on update and validates version.
    * @inheritdoc
    * @beta
    */
   protected static override onUpdate(arg: OnElementPropsArg): void {
     super.onUpdate(arg);
-    this.updateGeometry(arg.iModel, arg.props as TextAnnotation2dProps);
+    this.validateVersionAndUpdateGeometry(arg);
   }
 
   /**
    * Populates the `elementGeometryBuilderParams` property in the [TextAnnotation2dProps]($common).
-   * It only does this if the `elementGeometryBuilderParams` is not already set and if there is actually a text annotation to produce geometry for.
+   * Only does this if the `elementGeometryBuilderParams` is not already set and if there is actually a text annotation to produce geometry for.
+   * Also, validates the version of the text annotation data and migrates it if necessary.
+   * @beta
    */
-  protected static updateGeometry(iModelDb: IModelDb, props: TextAnnotation2dProps): void {
-    if (props.elementGeometryBuilderParams || !props.textAnnotationData) {
-      return;
+  protected static validateVersionAndUpdateGeometry(arg: OnElementPropsArg): void {
+    const props = arg.props as TextAnnotation2dProps;
+    const textAnnotationData = parseTextAnnotationData(props.textAnnotationData);
+    if (!props.elementGeometryBuilderParams && textAnnotationData) {
+      props.elementGeometryBuilderParams = getElementGeometryBuilderParams(arg.iModel, props.model, props.category, props.placement ?? Placement2d.fromJSON(), textAnnotationData.data, props.defaultTextStyle?.id);
     }
+  }
 
-    props.elementGeometryBuilderParams = getElementGeometryBuilderParams(iModelDb, props.model, props.placement ?? Placement2d.fromJSON(), props.textAnnotationData, props.category, props.defaultTextStyle ? props.defaultTextStyle.id : undefined);
+  /**
+   * TextAnnotation2d custom HandledProps include 'textAnnotationData'.
+   * @inheritdoc
+   * @beta
```

Comment:
```suggestion
   * @internal
```

Nobody needs to know you override this. Same for TextAnnotation3d.

Reply:
Done for `_customHandledProps`